bento/src/main/java/com/yelp/android/bento/core/AsyncInflationBridge.kt

```diff
@@ -0,0 +1,116 @@
+package com.yelp.android.bento.core
+
+import android.view.View
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.RecyclerView
+import com.yelp.android.bento.componentcontrollers.RecyclerViewComponentController.constructViewHolder
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.ConcurrentLinkedDeque
+import java.util.concurrent.Executors
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withContext
+
+/**
+ * This acts as a bridge between RecyclerViewComponentController and the underlying
+ * RecyclerView.Adapter. It intercepts the addComponent() calls and inflates views with
+ * [BentoAsyncLayoutInflater]. Components are added to the RecyclerView when the inflation
+ * is finished.
+ */
+internal class AsyncInflationBridge(val recyclerView: RecyclerView) {
+
+    private val context = recyclerView.context
```

Comment:
Do we need this to be a val? I think we only use it once

Reply:
If I pulled it inside the `coroutineScope` block I'd have to cast it a second time to know it's a `LifecycleOwner`.

Reply:
Ah gotcha. This is fine then.